### PR TITLE
Apply market output to getBuildingIO

### PR DESCRIPTION
**Pull request by jimmyh77**
getBuildingIO now calculates the output of markets.   Censor Office now who's the correct values if items are bought on markets.

Fixes https://github.com/fishpondstudio/CivIdle/issues/108

### CODE REVIEW

File: shared/logic/IntraTickCache.ts [hunk 1]
```diff
@@ -51,6 +51,18 @@ export function getBuildingIO(
...
+         const capacity = b.capacity * b.level * totalMultiplierFor(xy, "output", 1, gs);
+         forEach((b as IMarketBuildingData).sellResources, (k) => {
+            const buyResource = (b as IMarketBuildingData).availableResources[k]!;
+            result[buyResource] = round(
+               (capacity * getMarketPrice(k, xy, gs)) / getMarketPrice(buyResource, xy, gs),
+               1,
+            );
+         });
+         // Market output is not affected by multipliers
+         return result;
```
Comment (jimmyh77) [inline shared/logic/IntraTickCache.ts:64]: @insraq Should both of these set the cache before returning?

_cache.buildingIO.set(key, Object.freeze(result));

Comment (insraq) [reply shared/logic/IntraTickCache.ts:64]: Yes, cache should be set before returning :-)
It's not a big deal (potentially doing more calculations) but we should fix this just for consistency's sake.

File: shared/logic/IntraTickCache.ts [hunk 2]
```diff
@@ -51,6 +51,18 @@ export function getBuildingIO(
             resources[k] = 1;
          });
       }
+      if ("sellResources" in b && type === "output") {
+         const capacity = b.capacity * b.level * totalMultiplierFor(xy, "output", 1, gs);
+         forEach((b as IMarketBuildingData).sellResources, (k) => {
+            const buyResource = (b as IMarketBuildingData).availableResources[k]!;
+            result[buyResource] = round(
```
Comment (insraq) [inline shared/logic/IntraTickCache.ts:58]: The logic here feels like it must be implemented somewhere - we should probably resue that. (if there's no existing function, maybe create one)

Also usually we do not round in the logic (rounding is only done in the view layer)

File: shared/logic/IntraTickCache.ts (hunk 1 above)
Comment (jimmyh77) [reply shared/logic/IntraTickCache.ts:67]: Cache is now set on all returns

File: shared/logic/IntraTickCache.ts (hunk 2 above)
Comment (jimmyh77) [reply shared/logic/IntraTickCache.ts]: It was also in MarketBuildingBody.  Have move the code to a function in BuildingLOgic.ts

**Pull request closed without merging**